src/presetsController.ts

```diff
@@ -1025,7 +1025,12 @@ export class PresetsController {
 
         // Merge the includes in reverse order so that the final presets order is correct
         for (let i = presetsFile.include.length - 1; i >= 0; i--) {
-            const fullIncludePath = path.normalize(path.resolve(path.dirname(file), presetsFile.include[i]));
+            const rawInclude = presetsFile.include[i];
+            const includePath = presetsFile.version >= 7 ?
+                // Version 7 and later support $penv{} expansions in include paths
+                substituteAll(rawInclude, getParentEnvSubstitutions(rawInclude, new Map<string, string>())).result :
+                presetsFile.include[i];
```

Comment:
Nit: This could just refer to `rawInclude` couldn't it? This would avoid duplicate references.